### DISCUSSION TITLE
Possible Bug: RelationpManager uses accessor inside database query

### DISCUSSION THREAD

**Discussion by Timeswitch** [filamentphp/filament#4664] [Help]
When using an accessor as recordTitleAttribute inside a RelationManager, AttachAction adds it to the database query inside the ORDER BY clause.
This obviously fails.

Workaround:
Clear / override the queries ORDER BY clause:

```PHP
Tables\Actions\AttachAction::make()
                    ->recordSelectOptionsQuery(function (Builder $query) {
                        $query->getQuery()->orders = [];
                        $query->orders = [];
                        return $query;
                    })
```

Is this a bug or am I missing something?

**Comment by danharrin** [comment]
Maybe we need a `recordSelectSortColumns()` method to customise the sorting then, identical to `recordSelectSearchColumns()`. You can PR this, also to AssociateAction?
